tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -1500,3 +1503,38 @@ def test_store_ignores_hive_metastore_default_from_spark_session(mock_http, spar
     _verify_requests(
         mock_http, "registered-models/get", "GET", GetRegisteredModelRequest(name=name)
     )
+
+
+def test_store_use_presigned_url_store_when_disabled(monkeypatch):
+    monkeypatch.setenv(MLFLOW_UNITY_CATALOG_PRESIGNED_URLS_ENABLED.name, "False")
+    store_package = "mlflow.store._unity_catalog.registry.rest_store"
+
+    uc_store = UcModelRegistryStore(store_uri="databricks-uc", tracking_uri="databricks-uc")
+    model_version = ModelVersion(
+        name="catalog.schema.model_1", version="1", storage_location="s3://some/storage/location"
+    )
+    creds = TemporaryCredentials()
+    with mock.patch(
+        f"{store_package}.UcModelRegistryStore._get_temporary_model_version_write_credentials",
+        return_value=creds,
+    ) as temp_cred_mock, mock.patch(
+        f"{store_package}.get_artifact_repo_from_storage_info",
+    ) as get_repo_mock:
+        uc_store._get_artifact_repo(model_version)
+
+        temp_cred_mock.assert_called_once_with(
+            name=model_version.name, version=model_version.version
+        )
+        get_repo_mock.assert_called_once_with(
+            storage_location=model_version.storage_location, scoped_token=creds
+        )
+
+
+def test_store_use_presigned_url_store_when_enabled(monkeypatch):
+    monkeypatch.setenv(MLFLOW_UNITY_CATALOG_PRESIGNED_URLS_ENABLED.name, "True")
+    with mock.patch("mlflow.utils.databricks_utils.get_config"):
+        uc_store = UcModelRegistryStore(store_uri="databricks-uc", tracking_uri="databricks-uc")
+        model_version = ModelVersion(name="catalog.schema.model_1", version="1")
+        presigned_store = uc_store._get_artifact_repo(model_version)
+
+        assert type(presigned_store) is PresignedUrlArtifactRepository
```

Comment:
can we assert the type in the other test above as well?  Same with the 2nd time we test this down below.